### PR TITLE
Update comments and OpenZeppelin dependencies for CryptoPizza template

### LINKED ISSUE #165: Update CryptoPizza to use latest version of OpenZeppelin Contracts
## Summary

Update CryptoPizza to use latest version of OpenZeppelin Contracts. 
Currently using OpenZeppelin Contracts 2.3 when latest version is OpenZeppelin Contracts 2.4 
Also currently using `openzeppelin-solidity` when package is now `@openzeppelin/contracts`

## Motivation
Latest version is OpenZeppelin Contracts 2.4 (https://forum.openzeppelin.com/t/openzeppelin-contracts-v2-4/1665) and includes Gas Station Network support.

`openzeppelin-solidity` is now `@openzeppelin/contracts` see the announcement from July
https://forum.openzeppelin.com/t/announcing-the-openzeppelin-sdk/1092

## Additional context
Ideally update imports to not need to refer to `node_modules`:
`import "@openzeppelin/contracts/...` rather than:
`import "../node_modules/openzeppelin-solidity/contracts/...`

*Disclosure: I am the Community Manager at OpenZeppelin*

### PR DESCRIPTION
**Pull request by ChrisChinchilla**
This PR updates the CryptoPizza readme, and adds comments inline in the contract file. It also updates the OpenZeppelin dependency.

Also, I realised that I had parked adding instructions on the JavaScript as it was a big task. This is now in progress, but it will take a little while, so let's get this merged in the meantime.

Closes #165 amongst others (partially)

### CODE REVIEW

File: packages/editor/templates/Crypto Pizzas/README.md [hunk 1]
```diff
@@ -54,20 +54,20 @@ And that's all the code relevant to this dapp, there are other functions for glu
 > Find the CSS file in _app/app.css_
 > Find the JavaScript file in _app/app.js_
 
-### 1. Configure
+### Configure
 
-Configuring the contract allows you to set the name of the contract and the initial values sent to the constructor as arguments. You can configure the contract by clicking on the gear icon in the left panel.  In this example, the constructor doesn't accept arguments by default, so no configuration is necessary.
+Clicking the disclosure triangle next to the contract name reveals three sub-sections. In the _Configure_ sub-section you can change the name of the contract and set the initial values sent to the constructor arguments. In this example, the string displayed in the front end interface.
```
Comment (samajammin) [inline packages/editor/templates/Crypto Pizzas/README.md]: This is outdated, we don't want this change.

File: packages/editor/templates/Crypto Pizzas/contracts/CryptoPizza.sol [hunk 2]
```diff
@@ -1,89 +1,107 @@
+// Specifies that the source code is for a version
+// of Solidity greater than 0.5.10
 pragma solidity ^0.5.10;
 
-import "../node_modules/openzeppelin-solidity/contracts/token/ERC721/IERC721.sol";
-import "../node_modules/openzeppelin-solidity/contracts/token/ERC721/IERC721Receiver.sol";
-import "../node_modules/openzeppelin-solidity/contracts/introspection/ERC165.sol";
-import "../node_modules/openzeppelin-solidity/contracts/math/SafeMath.sol";
+// Imports symbols from other files into the current contract.
```
Comment (samajammin) [inline packages/editor/templates/Crypto Pizzas/contracts/CryptoPizza.sol]: I recommend adding a reference here (e.g. https://solidity.readthedocs.io/en/v0.6.2/layout-of-source-files.html#importing-other-source-files). I wasn't aware of what "symbols" meant... 😁

File: packages/editor/templates/Crypto Pizzas/contracts/CryptoPizza.sol [hunk 3]
```diff
@@ -1,89 +1,107 @@
...
+// In this case, a series of helper contracts from OpenZeppelin.
+import "../node_modules/@openzeppelin/contracts/token/ERC721/IERC721.sol";
+import "../node_modules/@openzeppelin/contracts/token/ERC721/IERC721Receiver.sol";
+import "../node_modules/@openzeppelin/contracts/introspection/ERC165.sol";
+import "../node_modules/@openzeppelin/contracts/math/SafeMath.sol";
 
+// A contract is a collection of functions and data (its state)
+// that resides at a specific address on the Ethereum blockchain.
+// "is" keyword inherits functions and  keywords from two external contracts
```
Comment (samajammin) [inline packages/editor/templates/Crypto Pizzas/contracts/CryptoPizza.sol]: Perhaps include reference to inheritance? https://solidity.readthedocs.io/en/v0.6.2/contracts.html#inheritance

File: packages/editor/templates/Crypto Pizzas/contracts/CryptoPizza.sol [hunk 4]
```diff
@@ -1,89 +1,107 @@
...
+    // You can nest mappings, this example maps owner to operator approvals
+    mapping(address => mapping(address => bool)) private operatorApprovals;
 
     // Create random Pizza from string (name) and DNA
-    function _createPizza(string memory _name, uint _dna)
+
+    // The internal visibility keyword means this function is
+    // only visible within this contract and contracts that derive this contract
+
+    // isUnique is a funciton modifier defined below that checks if the pizza
```
Comment (samajammin) [inline packages/editor/templates/Crypto Pizzas/contracts/CryptoPizza.sol]: `function`

File: packages/editor/templates/Crypto Pizzas/contracts/CryptoPizza.sol (hunk 2 above)
Comment (ChrisChinchilla) [reply packages/editor/templates/Crypto Pizzas/contracts/CryptoPizza.sol]: Kind of in the main read me, as then the file gets messy, an you can't click the links either, but for this particular one being so fundamental, done!

File: packages/editor/templates/Crypto Pizzas/contracts/CryptoPizza.sol (hunk 3 above)
Comment (ChrisChinchilla) [reply packages/editor/templates/Crypto Pizzas/contracts/CryptoPizza.sol]: It's kind of the same as the link above, but I reworded to be clearer

**Comment by ChrisChinchilla** [comment]
@samajammin Changes made

**Comment by lapolinar** [comment]
great job guys

File: packages/editor/templates/Crypto Pizzas/README.md [hunk 5]
```diff
@@ -42,9 +42,9 @@ The `createRandomPizza` function is the public function called in JavaScript tha
 
 The `generateRandomDna` introduces another new function modifier, [`pure`](https://solidity.readthedocs.io/en/v0.5.12/contracts.html#pure-functions). Pure functions promise not to read from or modify the state, instead they generally return values to another function that does.
 
-The `getPizzasByOwner` function is another public function called by JavaScript to return all pizzas created by the owner of a specified address. The function introduces another modifier, [`view`](https://solidity.readthedocs.io/en/v0.5.12/contracts.html#view-functions) which promise not to modify the state.
+The `getPizzasByOwner` function is another public function called by JavaScript to return all pizzas created by the owner of a specified address. The function introduces another modifier, [`view`](https://solidity.readthedocs.io/en/v0.5.12/contracts.html#view-functions) which promise not to modify the state. The function uses the [`memory`](https://solidity.readthedocs.io/en/latest/introduction-to-smart-contracts.html#storage-memory-and-the-stack) data storage area to only keep a valuable for the life of the contract instance.
```
Comment (samajammin) [inline packages/editor/templates/Crypto Pizzas/README.md]: I don't understand this sentence:
> The function uses the [`memory`](https://solidity.readthedocs.io/en/latest/introduction-to-smart-contracts.html#storage-memory-and-the-stack) data storage area to only keep a valuable for the life of the contract instance.

I assume you mean "value" vs. "valuable"? And could you please clarify what this means (for my understanding)? The life of a contract instance? Doesn't a contract instance last forever on the blockchain? Or is this contract instance a JavaScript object?

File: packages/editor/templates/Crypto Pizzas/README.md [hunk 6]
```diff
@@ -54,20 +54,20 @@ And that's all the code relevant to this dapp, there are other functions for glu
 > Find the CSS file in _app/app.css_
 > Find the JavaScript file in _app/app.js_
 
-### 1. Configure
+### Configure
```
Comment (samajammin) [inline packages/editor/templates/Crypto Pizzas/README.md]: Let's keep the numbering here to stay consistent with the other template READMEs

File: packages/editor/templates/Crypto Pizzas/README.md [hunk 7]
```diff
@@ -54,20 +54,20 @@ And that's all the code relevant to this dapp, there are other functions for glu
...
 
 Configuring the contract allows you to set the name of the contract and the initial values sent to the constructor as arguments. You can configure the contract by clicking on the gear icon in the left panel.  In this example, the constructor doesn't accept arguments by default, so no configuration is necessary.
 
-### 2. Compile
+### Compile
```
Comment (samajammin) [inline packages/editor/templates/Crypto Pizzas/README.md]: Same here

File: packages/editor/templates/Crypto Pizzas/README.md [hunk 8]
```diff
@@ -54,20 +54,20 @@ And that's all the code relevant to this dapp, there are other functions for glu
...
 
 Solidity is a compiled language, and you need to convert the Solidity code into bytecode before the contract can run. In the file tree under the contract file, you can compile the contract by clicking the _Compile_ sub-section, and output appears in the _Output_ pane.
 
-### 3. Deploy
+### Deploy
```
Comment (samajammin) [inline packages/editor/templates/Crypto Pizzas/README.md]: Same here

File: packages/editor/templates/Crypto Pizzas/README.md [hunk 9]
```diff
@@ -54,20 +54,20 @@ And that's all the code relevant to this dapp, there are other functions for glu
...
 
-Every smart contract runs at an address on the Ethereum blockchain, and you must deploy it to an address before it can run. When using Studio, the browser simulates the network, but there are several test networks and one main network for the Ethereum blockchain.
+Every smart contract runs at an address on the Ethereum blockchain, and needs an address before it can run. When using Studio, the browser simulates the network, but there are several test networks available, and one main network.
```
Comment (samajammin) [inline packages/editor/templates/Crypto Pizzas/README.md]: Why these changes? "needs an address before it can run" seems rather vague vs. the current "you must deploy it to an address before it can run". I also think the current wording "there are several test networks and one main network for the Ethereum blockchain." is more descriptive than your update of "there are several test networks available, and one main network."

File: packages/editor/templates/Crypto Pizzas/contracts/CryptoPizza.sol [hunk 10]
```diff
@@ -1,89 +1,108 @@
...
-
-        E.g. DNA 5142446803 - 51_Basis/42_Cheeses/44_Meats/68_Spices/03_Vegetables
-    */
-    uint constant dnaDigits = 10;
-    uint constant dnaModulus = 10 ** dnaDigits;
-
-    // Equals to `bytes4(keccak256("onERC721Received(address,address,uint256,bytes)"))`
-    // which can be also obtained as `IERC721Receiver(0).onERC721Received.selector`
+    // Constant variables in Solidity are similar to other languages
+    // but are not allowed to access storage, blockchain, or execution data,
```
Comment (samajammin) [inline packages/editor/templates/Crypto Pizzas/contracts/CryptoPizza.sol]: What do you mean that a constant variable is "not allowed to access storage"? These constant variables _are_ storage variables, aren't they?

File: packages/editor/templates/Crypto Pizzas/contracts/CryptoPizza.sol [hunk 11]
```diff
@@ -1,89 +1,108 @@
...
+    // or make calls to external contracts.
```
Comment (samajammin) [inline packages/editor/templates/Crypto Pizzas/contracts/CryptoPizza.sol]: Constant variables are not allowed to make calls to external contracts? I feel like that's obvious / confuses me. Why/how would a variable ever make a call to an external contract?

Comment (samajammin) [reply packages/editor/templates/Crypto Pizzas/contracts/CryptoPizza.sol]: I'd suggest a simpler definition of a constant variable... e.g. it's a read-only state variable (once set, you can't update it).

File: packages/editor/templates/Crypto Pizzas/README.md (hunk 9 above)
Comment (ChrisChinchilla) [reply packages/editor/templates/Crypto Pizzas/README.md]: In the time that has past I have no idea what my thinking was anymore… 🤔

**Comment by ChrisChinchilla** [comment]
@samajammin Updates pushed!

**Comment by javier-tarazaga** [comment]
@ChrisChinchilla the build was broken in this PR so you know
